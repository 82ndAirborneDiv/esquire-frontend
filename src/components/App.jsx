import React, { Component } from 'react';
import PropTypes  from 'prop-types';


// App's children
class App extends Component {
  render() {
    return (
      <div>
        La la la la
        { React.Children.toArray(this.props.children) }
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.arrayOf(PropTypes.node)
};

export default App;
