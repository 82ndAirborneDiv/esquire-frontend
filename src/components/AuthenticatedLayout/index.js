import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Grid from 'material-ui/Grid'
import Header from '../Header'

const mainStyles = {
  backgroundColor: '#f5f5f5',
  padding: '0 27px 37px 27px',
  flex: '1'
}

const AuthenticatedLayout = ({ user, children }) => {
  return (
    <Grid container spacing={0} direction="column" style={{ flex: '1' }}>
      <Header user={user} />
      <Grid container spacing={0} style={mainStyles}>
        {children}
      </Grid>
    </Grid>
    )
}

AuthenticatedLayout.propTypes = {
  user: PropTypes.object,
  children: PropTypes.node
}

const mapStateToProps = (state, props) => ({
  children: props.children,
  user: state.data.user.currentUser || {firstName: '', lastName: '', role: ''}
})

export default connect(mapStateToProps)(AuthenticatedLayout)
