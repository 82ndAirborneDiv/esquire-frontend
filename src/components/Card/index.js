import React from 'react'
import PropTypes from 'prop-types'
import Paper from 'material-ui/Paper'

const baseStyle = {
  flex: '1'
}

const Card = ({ children, style }) => {
  return (
    <Paper style={{ ...baseStyle, ...style }}>
      {children}
    </Paper>
  )
}

Card.propTypes = {
  children: PropTypes.node,
  style: PropTypes.object
}

export default Card