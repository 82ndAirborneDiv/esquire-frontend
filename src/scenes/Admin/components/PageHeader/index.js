import React from 'react'
import Grid from 'material-ui/Grid'
import Typography from 'material-ui/Typography'
import Button from 'components/Button'
import CircleIcon from 'components/CircleIcon'
import TextLink from 'components/TextLink'

const PageHeader = () => {
  return (
    <Grid container spacing={0} alignItems="center" style={{ height: '100px' }}>
      <Grid item xs>
        <Grid container spacing={8} alignItems="center">
          <Grid item>
            <CircleIcon circleColor="error" iconColor="white" circleSize="35px" iconSize="24px">face</CircleIcon>
          </Grid>
          <Grid item>
            <Typography type="headline">User Management</Typography>
          </Grid>
        </Grid>
      </Grid>
      <Grid item>
        <TextLink to="/admin/new/user" style={{ color: 'white' }}>
          <Button value=" + Add new user" color="accent" />
        </TextLink>
      </Grid>
    </Grid>
  )
}



export default PageHeader