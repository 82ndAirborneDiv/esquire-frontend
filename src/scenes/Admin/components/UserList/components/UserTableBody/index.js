import React from 'react'
import PropTypes from 'prop-types'
import { TableCell } from 'material-ui/Table'
import TableRow from 'components/TableRow'
import TextLink from 'components/TextLink'
import Typography from 'material-ui/Typography'

const styles = {
  padding: '5px',
  textAlign: 'center',
  fontWeight: 'lighter'
}

const Cell = ({ id, children, header }) => {
  return <TableCell key={`${id}-${header}`} style={styles} padding="none">{children}</TableCell>
}


const UserTableBody = ({ users }) => {
  return (
    users.map(user => (
      <TableRow key={user.id}>
        <Cell id={user.id} header="name">
          <span style={{ color: '#9a9c9c' }}>{user.firstName} {user.lastName}</span>
        </Cell>
        <Cell id={user.id} header="email">
          <span style={{ color: '#9a9c9c' }}>{user.email}</span>
        </Cell>
        <Cell id={user.id} header="role">
          <span style={{ color: '#9a9c9c', fontStyle: 'italic' }}>{user.role}</span>
        </Cell>
        <Cell id={user.id} header="edit">
          <TextLink to={'/admin/edit/user/' + user.id}>Edit</TextLink>
        </Cell>
      </TableRow>
    ))
  )
}

UserTableBody.propTypes = {
  users: PropTypes.arrayOf(PropTypes.object)
}

export default UserTableBody