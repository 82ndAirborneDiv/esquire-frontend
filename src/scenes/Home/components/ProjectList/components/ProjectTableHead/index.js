import React from 'react'
import PropTypes from 'prop-types'
import { TableSortLabel } from 'material-ui/Table'
import TableRow from 'components/TableRow'
import TableCell from 'components/TableCell'

const columns = [
  { key: 'bookmarked', label: '', style: { maxWidth: 10 }, hasSort: false },
  { key: 'name', label: 'Name', style: { textAlign: 'left', maxWidth: 'unset' }, hasSort: true },
  { key: 'dateLastEdited', label: 'Date Last Edited', style: { width: 150, maxWidth: 150, textAlign: 'unset' }, hasSort: true },
  { key: 'lastEditedBy', label: 'Last Edited By', style: { width: 150, maxWidth: 150, textAlign: 'unset' }, hasSort: true },
  { key: 'protocol', label: 'Protocol', hasSort: false },
  { key: 'jurisdictions', label: 'Jurisdictions', hasSort: false },
  { key: 'codingScheme', label: 'Coding Scheme', hasSort: false },
  { key: 'code', label: '', style: { maxWidth: 40 }, hasSort: false },
  { key: 'validate', label: '', style: { maxWidth: 40 }, hasSort: false },
  { key: 'export', label: 'Export', style: { maxWidth: 10 }, hasSort: false }
]

const hiddenCols = [
  'jurisdictions',
  'codingScheme',
  'validate'
]

const ProjectTableHead = ({ role, onRequestSort, sortBy, direction }) => {
  const visible = (role === 'Coder' ? columns.filter(c => !hiddenCols.includes(c.key)) : columns)
  return (
    <TableRow key="headers">
      {visible.map(c => (
        <TableCell key={c.key} style={c.style}>
          {c.hasSort ? (
            <TableSortLabel active={sortBy === c.key} direction={direction} onClick={onRequestSort(c.key)}>
              {c.label}
            </TableSortLabel>
          ) : c.label}
        </TableCell>
      ))}
    </TableRow>
  )
}

ProjectTableHead.propTypes = {
  role: PropTypes.string,
  onRequestSort: PropTypes.func,
  sortBy: PropTypes.string,
  direction: PropTypes.string
}

export default ProjectTableHead
