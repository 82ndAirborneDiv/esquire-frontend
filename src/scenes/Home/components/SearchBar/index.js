import React from 'react'
import PropTypes from 'prop-types'
import Autosuggest from 'react-autosuggest'
import match from 'autosuggest-highlight/match'
import parse from 'autosuggest-highlight/parse'
import Container, { Column, Row } from 'components/Layout'
import Icon from 'components/Icon'
import TextField from 'material-ui/TextField'

const renderInput = ({ autoFocus, value, ref, ...other }) => {
  return (
    <TextField/>
  )
}

const SearchBar = () => {
  return (
    <Container style={{ padding: '27px 0' }}>
      <Container alignItems="center" style={{ height: 57, backgroundColor: '#e5e5e5', padding: '0 15px' }}>
        <Column flex xs>
        </Column>
        <Column>
          <Icon color="primary">search</Icon>
        </Column>
      </Container>
    </Container>
  )
}

SearchBar.propTypes = {

}

export default SearchBar