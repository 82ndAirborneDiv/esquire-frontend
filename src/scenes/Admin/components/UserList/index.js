import React from 'react'
import PropTypes from 'prop-types'
import Grid from 'material-ui/Grid'
import Card from 'components/Card'
import Table from 'components/ListTable'
import UserTableBody from './components/UserTableBody'

const UserList = ({ users }) => {
  const headers = [
    { id: 'name', title: 'Name' },
    { id: 'email', title: 'Email' },
    { id: 'role', title: 'Role' },
    { id: 'edit', title: '' }
  ]

  return (
    <Grid container spacing={0} style={{ flex: '1', paddingTop: '25px' }}>
      <Grid item xs style={{ display: 'flex', flexDirection: 'column' }}>
        <Card>
          <Table headers={headers}>
            <UserTableBody users={users} />
          </Table>
        </Card>
      </Grid>
    </Grid>
  )
}

export default UserList