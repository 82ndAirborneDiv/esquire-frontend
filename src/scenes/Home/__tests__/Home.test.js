import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { MuiThemeProvider } from 'material-ui/styles'
import theme from 'services/theme'
import { Home } from '../'
import { PageHeader } from '../components/PageHeader'
import { ProjectList } from '../components/ProjectList'

let date = new Date('10/11/2017')
const projects = [
  { id: 1, bookmarked: true, name: 'Project 1', dateLastEdited: date, lastEditedBy: 'Kristin Muterspaw' },
  { id: 2, bookmarked: false, name: 'Project 2', dateLastEdited: date, lastEditedBy: 'Michael Ta' },
  { id: 3, bookmarked: true, name: 'Project 3', dateLastEdited: date, lastEditedBy: 'Sanjith David' },
  { id: 4, bookmarked: true, name: 'Project 4', dateLastEdited: date, lastEditedBy: 'Greg Ledbetter' },
  { id: 5, bookmarked: false, name: 'Project 5', dateLastEdited: date, lastEditedBy: 'Jason James' },
  { id: 6, bookmarked: false, name: 'Project 6', dateLastEdited: date, lastEditedBy: 'Hez Adedoyin' }
]

const props = {
  actions: {
    getProjectsRequest: jest.fn(),
    sortProjects: jest.fn(),
    updatePage: jest.fn(),
    updateRows: jest.fn(),
    toggleBookmark: jest.fn()
  },
  user: {
    id: 2,
    firstName: 'Kristin',
    lastName: 'Muterspaw',
    email: 'kmuterspaw@cdc.gov',
    password: 'test',
    role: 'Coordinator'
  },
  error: false,
  errorContent: '',
  projects,
  visibleProjects: projects,
  sortBy: 'name',
  direction: 'asc',
  page: 0,
  rowsPerPage: 10
}

const setup = otherProps => {
  return mount(
    <MemoryRouter>
      <MuiThemeProvider theme={theme}>
        <Home {...props} {...otherProps} />
      </MuiThemeProvider>
    </MemoryRouter>
  )
}

describe('Home scene', () => {
  test('it should render correctly', () => {
    expect(shallow(<Home {...props} />)).toMatchSnapshot()
  })

  test('it should render ProjectList and PageHeader components', () => {
    let wrapper = shallow(<Home {...props} />)

    expect(wrapper.find(ProjectList)).toHaveLength(1)
    expect(wrapper.find(PageHeader)).toHaveLength(1)
  })

  test('it should open the New Project modal when Create New Project is clicked', () => {
    let wrapper = setup()
    wrapper.find(PageHeader).find('Button').at(0).simulate('click')
    wrapper.update()
    expect(wrapper.find('Modal')).toHaveLength(1)
  })
})