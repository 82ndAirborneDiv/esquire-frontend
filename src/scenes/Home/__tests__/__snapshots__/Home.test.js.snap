// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home scene it should render correctly 1`] = `
<Container
  column={true}
  flex={true}
  spacing={0}
>
  <PageHeader
    role="Coordinator"
  />
  <withStyles(Divider) />
  <SearchBar />
  <ProjectList
    count={6}
    direction="asc"
    handlePageChange={[Function]}
    handleRequestSort={[Function]}
    handleRowsChange={[Function]}
    handleToggleBookmark={[Function]}
    page={0}
    projects={
      Array [
        Object {
          "bookmarked": true,
          "dateLastEdited": 2017-10-11T00:00:00.000Z,
          "id": 1,
          "lastEditedBy": "Kristin Muterspaw",
          "name": "Project 1",
        },
        Object {
          "bookmarked": false,
          "dateLastEdited": 2017-10-11T00:00:00.000Z,
          "id": 2,
          "lastEditedBy": "Michael Ta",
          "name": "Project 2",
        },
        Object {
          "bookmarked": true,
          "dateLastEdited": 2017-10-11T00:00:00.000Z,
          "id": 3,
          "lastEditedBy": "Sanjith David",
          "name": "Project 3",
        },
        Object {
          "bookmarked": true,
          "dateLastEdited": 2017-10-11T00:00:00.000Z,
          "id": 4,
          "lastEditedBy": "Greg Ledbetter",
          "name": "Project 4",
        },
        Object {
          "bookmarked": false,
          "dateLastEdited": 2017-10-11T00:00:00.000Z,
          "id": 5,
          "lastEditedBy": "Jason James",
          "name": "Project 5",
        },
        Object {
          "bookmarked": false,
          "dateLastEdited": 2017-10-11T00:00:00.000Z,
          "id": 6,
          "lastEditedBy": "Hez Adedoyin",
          "name": "Project 6",
        },
      ]
    }
    rowsPerPage={10}
    sortBy="name"
    user={
      Object {
        "email": "kmuterspaw@cdc.gov",
        "firstName": "Kristin",
        "id": 2,
        "lastName": "Muterspaw",
        "password": "test",
        "role": "Coordinator",
      }
    }
  />
  <Route
    component={[Function]}
    path="/new/project"
  />
</Container>
`;
